### PR TITLE
Allow Mask Processing in DFlash-v2 and Sampling Kernel

### PR DESCRIPTION
**Pull request by sempervictus**
Masking logits from beeing greedy sampled by the drafter reduces errant selection. The mechanism should also allow us to _adjust_ the draft sampler on the fly by generating compensatory masks and passing them into the new entrypoints.

### CODE REVIEW

**Comment by sempervictus** [comment]
Needed for https://github.com/guoqingbao/xinfer/pull/481 ( which incorporates https://github.com/guoqingbao/xinfer/pull/483 )

**Comment by sempervictus** [comment]
@guoqingbao - this PR aims to land atop #108 and at this point we know #108 works.
I'm sorting out a few issues with the TP implementation for DFlash (might be some geometric limitations on that, may need to fall back to full linear load in those cases) but the kernels in 108 are solid and this PR appears to work fine as well.

i added the un-spam thing to reduce eyestrain - i'm no good to you to debug stuff if i'm blind :-p

**Comment by sempervictus** [comment]
Replaced by #110
